### DISCUSSION TITLE
How could I set my Webcam to be the background?

### DISCUSSION THREAD

**Discussion by Tharavin** [rocksdanister/lively#342] [Q&A] [answered]
My Webcam monitors my front yard. I would like it to be my background (kind of like a virtual window).  I am new to lively so any suggestions how to create a wallpaper to do  this (or if its already been done)  would be appreciated.  I know VLC can display my webcam in full screen, but havent found a way to do that without going through the menu items every time I start it.

Thanks

**Comment by rocksdanister** [comment]
One way would be to make/use existing applications as wallpaper:
https://github.com/rocksdanister/lively/wiki/Application-Wallpaper

If its a stream link then you can pass it to mpv video player https://github.com/rocksdanister/lively/issues/341

**Comment by Tharavin** [comment]
unfortunately its not a stream, and I cant find an application that
automatically opens the webcam in fullscreen. in fact the only application
I have found that displays a webcam in full screen is VLC, but I cant find
a way to do it on startup, I have to manually set source to capture and
then set video playback to fullscreen.  I appreciate the response though
and Lively is a great project! I thought it would be an interesting and
potentially useful idea, but not necessary.

Thanks

**Reply by rocksdanister** [reply]
Its possible to start vlc with command line commands:
https://superuser.com/questions/1443481/vlc-record-from-webcam-and-save-to-file-windows-command-line
https://wiki.videolan.org/VLC_command-line_help

Once you get the right commands you can start it as application wallpaper https://github.com/rocksdanister/lively/issues/338#issuecomment-792223895

Unfortunately I can't help you myself as I don't have a webcam to test.

**Comment by Tharavin** [comment]
Thanks, I'll play around with it when I get home

**Comment by Tharavin** [comment]
Ok, I finally got something working. created LivelyInfo.json as follows:
{
  "AppVersion": "1.2.0.2",
  "Title": "Window",
  "Thumbnail": null,
  "Preview": null,
  "Desc": null,
  "Author": null,
  "License": null,
  "Contact": null,
  "Type": 0,
  "FileName": "C:\\Program Files (x86)\\VideoLAN\\VLC\\vlc.exe",
  "Arguments": "dshow:// :dshow-vdev=\"c922 Pro Stream Webcam\"
:dshow-aspect-ratio=\"16:9\"  --fullscreen",
  "IsAbsolutePath": true
}

zipped it into Window.zip, then clicked on plus sign in Lively Wallpaper,
then Browsed to it and it was imported into library.

There's probably an easier way to import it, but dragging didn't work for
json or zip for me, and browsing to json directly did not work.

I'll probably touch it up and add a thumbnail/preview at some point, but
for now I'm just glad it works!

Thanks again for the help, especially the pointer to the chrome example!

**Comment by rocksdanister** [comment]
Great!

I played around a bit with vlc and video files, I would make the following changes to the Arguments:
```json
{
  "AppVersion": "1.2.0.2",
  "Title": "Window",
  "Thumbnail": null,
  "Preview": null,
  "Desc": null,
  "Author": null,
  "License": null,
  "Contact": null,
  "Type": 0,
  "FileName": "C:\\Program Files\\VideoLAN\\VLC\\vlc.exe",
  "Arguments": "\"H:\\Vid\\test.mp4\" --no-qt-video-autoresize --no-qt-system-tray --qt-minimal-view",
  "IsAbsolutePath": true
}
```
**--qt-minimal-view :** hide menus and controls.
**--no-qt-video-autoresize :** prevent player window resizing to video size.
**--no-qt-system-tray :** do not create system-tray icon.

`--fullscreen` needs to be removed because resizing and positioning of the window should be handled by Lively itself and not the application.

**Comment by Tharavin** [comment]
Thanks, I'll check it out.

**Comment by Tharavin** [comment, marked as answer]
here's what I ended with.  works great!

{
  "AppVersion": "1.2.0.2",
  "Title": "WindowCam",
  "Thumbnail": null,
  "Preview": null,
  "Desc": null,
  "Author": null,
  "License": null,
  "Contact": null,
  "Type": 0,
  "FileName": "C:\\Program Files (x86)\\VideoLAN\\VLC\\vlc.exe",
  "Arguments": "dshow://  :dshow-vdev=\"c922 Pro Stream Webcam\"
:dshow-adev=none  :dshow-aspect-ratio=\"16:9\" :live-caching=300
--no-qt-video-autoresize --no-qt-system-tray --qt-minimal-view",
  "IsAbsolutePath": true
}

Thanks for all your help!
